oop2-encapsulate-inherit.js

```diff
@@ -5,41 +5,291 @@
     - Site class. It should have the name of the company, and the location of the site.  It is a child class that inherits from the Company class.  Create 3 sites for each company.
     - Employee class. It should have a employee name, job title, and salary properties. Create 10 employee instances: 1 CEO, 1 Manager, 1 Secretary, 2 Engineers, 1 Financial Officer, 1 Janitor, 1 Marketer, 1 HR Personnel, 1 Lawyer.  Feel free to add other employee roles at your own discretion.
     
     - For each class, add 2 properties and 2 methods to each.  The properties and methods should make sense, considering what the classes are supposed to be representing.
-
+*/
 
 
     //your code here...
 
+class Umbrella {
+    #orgName;
+    #yearFormed = 1999;
+    #listOfCompanies = ['TEKsystems', 'Aerotek', 'Aston Carter'];
+    #updatedList;
+    constructor(orgName) {
+        this.#orgName = orgName;
+    }
+    acquireCompany(compName) {
+        if(this.#listOfCompanies.includes(compName)) {
+            console.log(`${compName} is already in the Allegis Group, and therefore cannot be acquired.`)
+        } else {
+            this.#listOfCompanies.push(compName)
+        }
+        return this.#listOfCompanies;
+    }
+    consolidateCompany(compName) {
+        if(this.#listOfCompanies.includes(compName)) {
+            this.#updatedList = this.#listOfCompanies.filter(function(company) {return company !== compName})
+        } else {
+            console.log(`${compName} is not in the Allegis Group, and therefore cannot be consolidated.`)
+        }
+        return this.#updatedList;
+    }
+}
+
+class Company extends Umbrella {
+    #compName;
+    #employeeCount;
+    #brandColors;
+    #yearFounded;
+    constructor(compName, employeeCount, brandColors, yearFounded) {
+        super()
+        this.#compName = compName;
+        this.#employeeCount = employeeCount;
+        this.#brandColors = brandColors;
+        this.#yearFounded = yearFounded;
+    }
+    reBrand(colorsArray) {
+        return this.#brandColors = colorsArray;
+    }
+    makeAnnouncement() {
+        console.log(`${this.#compName} is now offering a new benefits package to all ${this.#employeeCount} employees!`)
+    }
+}
+
+class Site extends Company {
+    #location;
+    #compName;
+    #inBusiness = true;
+    #isIndoor = true;
+    constructor(compName, location) {
+        super(compName);
+        this.#compName = compName;
+        this.#location = location;
+    }
+    hostEvent(eventName) {
+        console.log(`${this.#compName} is hosting ${eventName} at their ${this.#location} location!`)
+    }
+    closeLocation() {
+        console.log(`${this.#compName} is closing their ${this.#location} location, permanently.`)
+        this.#inBusiness = false;
+    }
+    getInBusinessStatus() {
+        return this.#inBusiness;
+    }
+}
+
+class Employee extends Site {
```

Comment:
inheritance vs composition...  is an Employee a Site, Company or Umbrella?
(parent objects should probably have an array of components, rather than inheritance)